### PR TITLE
in createUser, create a corresponding entry in the userImages table

### LINKED ISSUE #227: Reading and writing userImages is broken
Somewhere in the chaos, we've lost the ability to read/write to our `userImages` collection:

We are getting an error like:
```
Error updating image FirebaseError: Missing or insufficient permissions.
    at new n (http://localhost:8080/dist/app.js:21807:23)
    at http://localhost:8080/dist/app.js:36721:34
    at Y.<anonymous> (http://localhost:8080/dist/app.js:36676:21)
    at ub (http://localhost:8080/dist/app.js:39857:22)
    at Y../node_modules/@firebase/webchannel-wrapper/dist/index.esm.js.g.dispatchEvent (http://localhost:8080/dist/app.js:39837:50)
    at Z../node_modules/@firebase/webchannel-wrapper/dist/index.esm.js.Z.ma (http://localhost:8080/dist/app.js:40755:40)
    at nc (http://localhost:8080/dist/app.js:40125:138)
    at qc (http://localhost:8080/dist/app.js:40027:9)
    at H../node_modules/@firebase/webchannel-wrapper/dist/index.esm.js.g.ya (http://localhost:8080/dist/app.js:40002:35)
    at H../node_modules/@firebase/webchannel-wrapper/dist/index.esm.js.g.Sa (http://localhost:8080/dist/app.js:39967:84)
```

For both reads and writes

### PR DESCRIPTION
**Pull request by ibeckermayer**
Figured out that the `userImages` error was due to each user not having a corresponding document in the `userImages` table. 
- Updated the `createUser` endpoint to automatically create an entry each time a user is created
- Updated tests to check for it
- Updated and cleaned up `reset-infrastructure.js`

closes #227

### CODE REVIEW

File: cloud-functions/functions/reset-infrastructure.js [hunk 1]
```diff
@@ -28,244 +28,140 @@ async function addMinimalSampleData() {
...
-      } catch (err_7) {
-        throw err_7;
-      }
-    } catch (err_8) {
-      throw err_8;
-    }
-  } catch (err_9) {
-    throw err_9;
-  }
+async function createUser(user) {
```
Comment (aidan-fitz) [inline cloud-functions/functions/reset-infrastructure.js:93]: I like this refactor!

File: cloud-functions/functions/reset-infrastructure.js [hunk 2]
```diff
@@ -28,244 +28,140 @@ async function addMinimalSampleData() {
...
+  await db.collection('users').doc(user.email).set({
+    isAdmin: user.isAdmin,
+    organizationID: user.organizationID,
+    disabled: user.disabled,
+    firstName: user.firstName,
+    lastName: user.lastName,
+    isFirstTimeUser: user.isFirstTimeUser,
+  });
+  await db.collection('userImages').doc(user.email).set({});
```
Comment (aidan-fitz) [inline cloud-functions/functions/reset-infrastructure.js:102]: Should probably be consistent with the change you made in `cloud-functions/functions/src/index.ts`:

```suggestion
  await db.collection('userImages').doc(user.email).set({ imageBlob: null });
```

File: cloud-functions/functions/reset-infrastructure.js [hunk 3]
```diff
@@ -28,244 +28,140 @@ async function addMinimalSampleData() {
...
+  await initechRef.set({
+    name: 'Initech',
+  });
+  await createUser({
+    email: 'admin@soylentgreen.com',
+    isAdmin: true,
+    organizationID: soylentGreenID,
+    disabled: false,
+    firstName: uniqueNamesGenerator({ dictionaries: [names], length: 1 }),
+    lastName: uniqueNamesGenerator({ dictionaries: [names], length: 1 }),
```
Comment (aidan-fitz) [inline cloud-functions/functions/reset-infrastructure.js:43]: Can we move the `firstName: ... lastName: ...` lines into `createUser()`? They are the same for every invocation.

Comment (ibeckermayer) [reply cloud-functions/functions/reset-infrastructure.js:43]: Not worth it imo, this implementation is already built and more flexible.

Comment (aidan-fitz) [reply cloud-functions/functions/reset-infrastructure.js:43]: Okay.

**Pull request merged**